Chessington.GameEngine/Board.cs

```diff
@@ -59,6 +59,17 @@ public void MovePiece(Square from, Square to)
 
             //Move the piece and set the 'from' square to be empty.
             board[to.Row, to.Col] = board[from.Row, from.Col];
+
+            var playerColor = board[to.Row, to.Col].Player;
+
+            if(playerColor == Player.Black && board[to.Row, to.Col].GetType() == typeof(Pawn) && to.Row == 7)
+            {
+                board[to.Row, to.Col] = new Queen(board[to.Row, to.Col].Player);
+            }
+            else if (playerColor == Player.White && board[to.Row, to.Col].GetType() == typeof(Pawn) && to.Row == 0)
+            {
+                board[to.Row, to.Col] = new Queen(board[to.Row, to.Col].Player);
+            }
```

Comment:
You can extract this into a `PromotePawns()` method, since that's an independent logic unit.